python/cudf/cudf/dataframe/multiindex.py

```diff
@@ -282,34 +308,66 @@ def _get_row_major(self, df, row_tuple):
                 result = result[result.columns[0]]
                 # convert to Series
                 series_name = []
-                for idx, code in enumerate(result.columns.codes):
-                    series_name.append(
-                        result.columns.levels[idx][
-                            result.columns.codes[code][0]
-                        ]
-                    )
+                for idx, code in enumerate(index._source_data.columns):
+                    series_name.append(result.columns._source_data[code][0])
                 result = Series(
-                    list(result._cols.values())[0], name=series_name
+                    list(result._cols.values())[0],
+                    name=series_name,
+                    index=result.index,
                 )
                 result.name = tuple(series_name)
+            elif len(result) == 0 and len(result.columns) == 0:
+                if slice_access is False:
+                    # Pandas returns a Series with a stringified index for
+                    # the one expected result column
+                    series_name = []
+                    for idx, code in enumerate(index._source_data.columns):
+                        series_name.append(index._source_data[code][0])
+                    result = Series([], name=series_name)
+                    result.name = tuple(series_name)
             elif (len(out_index.columns)) > 0:
                 # Otherwise pop the leftmost levels, names, and codes from the
                 # source index until it has the correct number of columns (n-k)
```

Comment:
stale comment?

Reply:
I refactored this block and improved the commenting.